### PR TITLE
 Respect default excludes for snapshotting

### PR DESCRIPTION
**Pull request by wolfs**
We dropped respecting the default excludes for snapshotting when we converted to hierarchical snapshots.

### CODE REVIEW

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy [hunk 1]
```diff
@@ -269,4 +269,35 @@ class TaskUpToDateIntegrationTest extends AbstractIntegrationSpec {
         then:
         executedAndNotSkipped ":customTask"
     }
+
+    def "changing default excludes leaves task up-to-date"() {
```
Comment (lptr) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy]: The test name suggests it's the default excludes you are changing, I think `"changes to inputs that are excluded by default leave task up-to-date"` is a better name.

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy [hunk 2]
```diff
@@ -269,4 +269,35 @@ class TaskUpToDateIntegrationTest extends AbstractIntegrationSpec {
...
+        run 'myTask'
+        then:
+        executedAndNotSkipped(':myTask')
+
+        when:
+        inputDir.file('.gitignore').text = "some ignored file"
+        inputDir.file('#ignored#').text = "some ignored file"
+        inputDir.file('.git/any-name.txt').text = "some ignored file"
+        inputDir.file('._ignored').text = "some ignored file"
+        inputDir.file('some-file.txt~').text = "some ignored file"
```
Comment (lptr) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy]: Can we try something like `inputDir.file("something/.git/any-name.txt")`?

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy (hunk 1 above)
Comment (wolfs) [reply subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy]: Changed.

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy (hunk 2 above)
Comment (wolfs) [reply subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy]: Added.

**Pull request merged**